### ISSUE TITLE
The JSON value of length n is too large and not supported #2

### ISSUE THREAD

**Issue by albracko** [dotnet/runtime#39953]
I'm referring to this issue https://github.com/dotnet/runtime/issues/30746 that was closed with limit of 125MB staying fixed opposed to being configurable.

It was argued that there would be no common cases hitting the 125MB limit. Such cases probably never occur on UI but there are such cases...for example if you have a REST API that returns JSON document which has some document encoded as a base64 in a single field or it has another JSON document nested inside a single field as a string which is used for digital signing and verification of it. 
And i have such a case where i'm digitally signing some JSON document and the creating a new JSON where the payload field (contains my JSON document) is larger than the 125MB limit.

**Comment by eiriktsarpalis** [comment]
Agree that this should likely be made configurable.

**Comment by minhtuanit** [comment]
Any update with this issue?

**Comment by inf9144** [comment]
This hardcoded thing just crashed my production site. Fix ahead or do i need a new serializer?

**Comment by albracko** [comment]
Yeah, sadly for the time being, the only solution is to use Newtonsoft for big JSON fields

**Comment by kasperk81** [comment]
> Agree that this should likely be made configurable.

instead of making it configurable, how about remove the limit? let it scale with system's available resources and let the stack overflow in worse case.

**Comment by albracko** [comment]
i'm for any of those 2 options as long as it doesn't have the hard limit

**Comment by Luk164** [comment]
Jusy hit the issue as well, wanted to serialize a dataset and it was just over the hard limit.

**Comment by eiriktsarpalis** [comment]
Related issue: #61089

**Comment by osexpert** [comment]
"640K ought to be enough for anybody."

**Comment by albracko** [comment]
Sorry, but if YOU don't need it that doesn't automatically mean everybody else doesn't need it.
YOU is not ANYBODY/EVERYBODY ;)

**Comment by krwq** [comment]
Would any of these two help here?
- https://github.com/dotnet/runtime/issues/78159
- https://github.com/dotnet/runtime/issues/67337

**Comment by albracko** [comment]
None of those would help in my case since they are both talking about streaming. I on the other had need to serialize the object into a JSON string which i also have to store. 
If it were just a case of returning this JSON through REST API, then yeah, those 2 issues that are talking about streaming would help.

**Comment by albracko** [comment]
Not really, if i'm not mistaken both of them are using streaming as a solution. But i'm not streaming the contents across network, i'm trying to store the serialized JSON into database.

**Comment by AlanMacdonald** [comment]
@stephentoub this is pretty painful. I just got this on a dotnet 6 production system with a background job that processes customer documents that calls out to other internal APIs for some work and uses the httpClient.PostAsJsonAsync method. In this case the customer document was huge and way above average size. You had comments on https://github.com/dotnet/corefx/pull/40792/files#r320546088 that sounded like you were hoping these limits could be removed in a future version.

It is very counterintuitive to hit an arbitrary limit in an otherwise enterprise level framework. I will need to look into whether I can change the serializer to NewtonSoft and keep these http client calls the same or change the code to not use them so we can control the serializer or look into alternatives to handle rare cases of extremely large documents.

It would be great if the limit here was available memory and not an arbitrary restriction.

**Comment by eiriktsarpalis** [comment]
Reopening to track some of the other hardcoded limits such as the following:

https://github.com/dotnet/runtime/blob/39e022d338e3d78c79b938aef8b14201a4546ab6/src/libraries/System.Text.Json/src/System/Text/Json/JsonConstants.cs#L71-L73

**Comment by EmanueleBaron** [comment]
Good night!!

Has the limit in json been fixed in the latest version in System.Text.Json? And that?

**Comment by eiriktsarpalis** [comment]
If you're asking about the fix in https://github.com/dotnet/runtime/pull/85334, it should be available with .NET 8. You can try it out today using the RC2 NuGet package: https://www.nuget.org/packages/System.Text.Json/8.0.0-rc.2.23479.6

**Comment by sxotney** [comment]
I'm not sure if it's resolved or I'm missing something. I came across this problem in .NET 8.0 GA. My use case being users of our systems wish to download a csv format of their data from our Blazor based reporting tool. These data downloads could easily stretch to over 125mb

I send a base64 string representation of the csv to the JSRuntime and a download is then triggered via a javascript call. It's a neat solution that I'd prefer to keep and can't think of any workarounds due to the JSRuntime dependency.

**Comment by Luk164** [comment]
> I'm not sure if it's resolved or I'm missing something. I came across this problem in .NET 8.0 GA. My use case being users of our systems wish to download a csv format of their data from our Blazor based reporting tool. These data downloads could easily stretch to over 125mb
> 
> I send a base64 string representation of the csv to the JSRuntime and a download is then triggered via a javascript call. It's a neat solution that I'd prefer to keep and can't think of any workarounds due to the JSRuntime dependency.
> 

Honestly for that case it might be better to instead generate a file in multiple steps and download that, this issue has been with us for way too long to wait for it to get resolved

**Comment by rohkhann** [comment]
What is the defined limit for S for Deserialization of a json string into a json document? I see the max limit be defined as : 166_666_666 for MaxCharacterTokenSize but in the fix in this pr:[#85534](https://github.com/dotnet/runtime/pull/85334) it seems to be Int.MaxValue/4 * 3

**Comment by shargon** [comment]
That my case, this should be configurable

**Comment by shargon** [comment]
In my case seems that there is a kind of race condition, sometimes it works, and when it start to fail, already fails for the next times, always with the same data

**Comment by neharastogi** [comment]
We're also encountering this 125MB limit in production. Our use case involves serializing JSON responses that contain base64-encoded document content as string properties (not byte arrays). When the encoded content exceeds the limit, serialization fails entirely.                                                                                                                                                                                                   
                                                                                                                                                                                                                                         This is particularly problematic for document-heavy APIs where embedding document content as base64 strings within JSON responses is a standard pattern required by industry specifications.                                           
                                                                                                                                                                                                                                         
 A configurable limit or removal of this restriction would be very welcome. Adding another voice here to indicate this is a real-world production issue, not just an edge case.

**Comment by osexpert** [comment]
Would it be possible to create a jsonstream that would be limitless?
